frontend/src/main/scala/bloop/data/SourcesGlobs.scala

```diff
@@ -0,0 +1,69 @@
+package bloop.data
+
+import java.nio.file.FileSystems
+import java.nio.file.PathMatcher
+import java.nio.file.Path
+import bloop.config.Config
+import scala.util.Properties
+
+case class SourcesGlobs(
+    directory: Path,
+    walkDepth: Int,
+    includes: List[PathMatcher],
+    excludes: List[PathMatcher]
+) {
+  def matches(path: Path): Boolean = {
+    def matchesList(lst: List[PathMatcher]): Boolean = lst match {
+      case Nil => false
+      case head :: tail =>
+        if (head.matches(path)) true
+        else matchesList(tail)
+    }
+    matchesList(includes) && !matchesList(excludes)
+  }
+}
+
+object SourcesGlobs {
+  def fromConfig(project: Config.Project): List[SourcesGlobs] = {
+    project.sourcesGlobs match {
+      case None => Nil
+      case Some(globs) =>
+        globs.flatMap { glob =>
+          SourcesGlobs.fromStrings(
+            glob.directory,
+            glob.walkDepth,
+            glob.includes,
+            glob.excludes
+          )
+        }
+    }
+  }
+
+  def fromStrings(
+      directory: Path,
+      walkDepth: Option[Int],
+      includes: List[String],
+      excludes: List[String]
+  ): List[SourcesGlobs] = {
+    if (includes.isEmpty) Nil
+    else {
+      val uriPath = directory.toUri().getPath()
+      val basePath =
+        if (Properties.isWin) uriPath.stripPrefix("/")
+        else uriPath
+      val prefix = s"glob:${basePath}"
```

Comment:
It would be nice to avoid the prefix here for (hypothetically) performance reasons but I struggled to get it working correctly

<img width="1106" alt="Screenshot 2020-02-14 at 18 53 01" src="https://user-images.githubusercontent.com/1408093/74559002-45ce7d00-4f5b-11ea-930b-00bd3e12bd33.png">

I expected `glob:*Test.scala` to match `/Users/lgeirsson/FooTest.scala` 🤔

Reply:
I figured out how to avoid the absolute globs. The trick was the relativize the path before calling `matches()`, which makes sense.